planner/core/task.go

```diff
@@ -682,6 +683,39 @@ func (p *basePhysicalAgg) newPartialAggregate() (partial, final PhysicalPlan) {
 		groupByItems = append(groupByItems, gbyCol)
 	}
 
+	// Remove unnecessary schema column.
+	// When the select column is same with the group by key, the column can be removed and gets value from the group by key.
+	// e.g
+	// select a, count(b) from t group by a;
+	// The schema is [firstrow(a), count(b), a]. The column firstrow(a) is unnecessary.
+	// Can optimize the schema to [count(b), a] , and change the index to get value.
+	partialCursor = 0
```

Comment:
This function is a little too long, how about extracting a function for it, for example `removeUnnecessaryFirstRow()`?

Reply:
Update.  And I add a function `removeUnnecessaryFirstRow()`